### PR TITLE
Fix Inkeep search modal state management and keyboard shortcuts

### PR DESCRIPTION
**Pull request by mosch**
## Summary

Fixes the Inkeep search plugin to properly manage the modal's open/closed state and keyboard shortcuts. The modal is now controlled by Zudoku rather than Inkeep, ensuring that user interactions (keyboard shortcuts, button clicks, escape key) correctly update the application state.

## Key Changes

- **State Management**: Modified `InkeepSearch` component to accept both `onOpen` and `onClose` callbacks, mirroring state changes in both directions. The modal is now fully controlled through the `isOpen` prop, so Inkeep can only report what it wants to do via `onOpenChange`.

- **Keyboard Shortcut Handling**: Disabled Inkeep's built-in `shortcutKey` by default (set to `null`) since Zudoku owns the ⌘K / Ctrl+K shortcut through the search button. This prevents Inkeep from swallowing the keyboard event before Zudoku's listener can handle it. Users can still configure an additional shortcut via `modalSettings.shortcutKey`.

- **Widget Lifecycle**: Moved widget instantiation into a `useEffect` to ensure only one widget instance is mounted across re-renders, preventing stray event listeners. Added polling logic to handle deferred Inkeep script loading.

- **API Update**: Added `onOpen` parameter to the `SearchProviderPlugin.renderSearch` interface so plugins can respond to both open and close events.

- **Documentation**: Updated search configuration docs to clarify that Zudoku manages the modal state and owns the default keyboard shortcut.

## Implementation Details

- The Inkeep widget uses Radix's `useControllableState` internally, so once `modalSettings.isOpen` is controlled, the widget can no longer open/close itself—it can only report requests via `onOpenChange`.
- Both directions of state change must be mirrored back into Zudoku's state: when Inkeep wants to open (via shortcut or trigger), and when it wants to close (via Escape or other means).
- Comprehensive test suite added covering keyboard shortcuts, button clicks, state synchronization, and edge cases like reopening after close.

https://claude.ai/code/session_01BCP7CR2yENSNLTZkaevTrU

### CODE REVIEW

File: packages/zudoku/src/lib/components/Search.tsx [hunk 1]
```diff
@@ -55,7 +55,9 @@ export const Search = ({ className }: { className?: string }) => {
           <KbdShortcut />
         </ClientOnly>
       </button>
-      <Suspense>{searchPlugin.renderSearch({ isOpen, onClose })}</Suspense>
+      <Suspense>
+        {searchPlugin.renderSearch({ isOpen, onOpen, onClose })}
```
[the remark this answers is not in the document]
Comment (mosch) [reply packages/zudoku/src/lib/components/Search.tsx]: Confirmed and fixed in 48a095e, taking the approach you described.

The open state, the ⌘K listener and `renderSearch` now live in a `SearchProvider` that `Header` renders once; `Search` is just the button and pulls `onOpen` from context. The shortcut listener is also skipped when no search plugin is configured, rather than calling `preventDefault()` for a modal that never opens.

Worth noting the defect was provider-agnostic and predates this PR — Pagefind stacked two dialogs the same way — so this fixes both.

`Header.test.tsx` now covers it: parametrized over the `start`/`center`/`end` placements asserting ⌘K opens exactly one modal, plus a case that every rendered search button opens the same one. Against the previous code, `start` and `end` both produced 2 modals and the responsive button opened none.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/zudoku/src/lib/plugins/search-inkeep/index.tsx [hunk 2]
```diff
@@ -83,24 +94,28 @@ const InkeepSearch = ({
...
+    // effect and never during render.
+    const createInstance = () => {
+      const instance = window.Inkeep?.ModalSearchAndChat?.(config);
+      if (instance) setSearchInstance(instance);
+
+      return Boolean(instance);
+    };
+
+    // The Inkeep script is loaded deferred, so poll until it is available
+    if (createInstance()) return;
```
[the remark this answers is not in the document]
Comment (mosch) [reply packages/zudoku/src/lib/plugins/search-inkeep/index.tsx]: Confirmed and fixed in cfc2589. `StrictMode` is indeed on at `Bootstrap.tsx:38`, and the setup/cleanup/setup cycle left the first widget mounted while state pointed at the second — so the widget being controlled was not the only one on the page.

The instance is now local to the effect with cleanup always returned, and `searchInstance` is off the dependency list.

One adjustment to the suggestion: the teardown is deferred with `setTimeout`, following `GraphiQLViewer.tsx:150`, since `unmount()` tears down a nested React root and doing that synchronously from an effect cleanup unmounts a root mid-render.

Covered by three tests that fail against the previous code. The StrictMode one is the clearest: widget open state came back `[false, true]` — two widgets, with the stale one left closed.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by vazexqi** [comment]
Reviewed and Approved
